source/opt/vector_dce.cpp

```diff
@@ -351,13 +355,19 @@ void VectorDCE::AddItemToWorkListIfNeeded(
     WorkListItem work_item, VectorDCE::LiveComponentMap* live_components,
     std::vector<WorkListItem>* work_list) {
   ir::Instruction* current_inst = work_item.instruction;
-  auto it = live_components->find(current_inst->result_id());
-  if (it == live_components->end()) {
-    live_components->emplace(
-        std::make_pair(current_inst->result_id(), work_item.components));
+
+  if (current_inst->result_id() >= live_components->size()) {
```

Comment:
There's an opportunity to cache the result id. ... but that might be a bad micro-optimization.
This is fine.